### ISSUE TITLE
Use non-`torch` loss functions

### ISSUE THREAD

**Issue by Pale-Blue-Dot-97** [Pale-Blue-Dot-97/Minerva#27]
### Issue
Currently, only loss functions from the `torch.nn` module can be selected via `YAML`. But there are loss functions from other projects (such as `pytorch-metric-learning`) that would be particularly useful.

### Solution

- [x] Modify `Trainer.make_criterion` so that the module name for the loss function is supplied to `utils.func_by_str` and not just hard-coded to `torch.nn`.
- [x] Updated `config` `YAML` files to include a `module` field with the loss params.
